src/solvers/refinement/string_refinement.cpp

```diff
@@ -311,33 +311,81 @@ bool is_char_pointer_type(const typet &type)
 }
 
 /// \param type: a type
-/// \param ns: name space
-/// \return true if a subtype is an pointer of characters
-static bool has_char_pointer_subtype(const typet &type, const namespacet &ns)
+/// \param pred: a predicate
+/// \return true if one of the subtype of `type` satisfies predicate `pred`.
+///         The meaning of "subtype" is in the algebraic datatype sense:
+///         for example, the subtypes of a struct are the types of its
+///         components, the subtype of a pointer is the type it points to,
+///         etc...
+///         For instance in the type `t` defined by
+///         `{ int a; char[] b; double * c; { bool d} e}`, `int`, `char`,
+///         `double` and `bool` are subtypes of `t`.
+bool has_subtype(
+  const typet &type, const std::function<bool(const typet &)> &pred)
 {
-  if(is_char_pointer_type(type))
+  if(pred(type))
     return true;
 
+  // clang-format off
```

Comment:
I think we have been so far using clang-formats style where it disagrees with the linter (And using `// NOLINTNEXTLINE`)